crates/symbolicator-service/src/services/sourcemap_lookup.rs

```diff
@@ -369,6 +244,54 @@ impl SourceMapLookup {
         cached_sourcesmap
     }
 
+    /// Fetches an arbitrary file using its `abs_path`,
+    /// or optionally its [`DebugId`] and [`SourceFileType`]
+    /// (because multiple files can share one [`DebugId`]).
+    pub async fn get_file(&mut self, key: FileKey) -> CacheEntry<CachedFile> {
+        // First, we do a trivial lookup in case we already have this file.
+        // The `abs_path` uniquely identifies a file in a JS stack trace, so we use that as the
+        // lookup key throughout.
+        if let Some(file) = self.files_by_key.get(&key) {
+            return file.clone();
+        }
+
+        // Try looking up the file in one of the artifact bundles that we know about.
+        if let Some(file) = self.try_get_file_from_bundles(&key) {
+            return self.files_by_key.entry(key).or_insert(Ok(file)).clone();
+        }
+
+        // Otherwise, try to get the file from an individual artifact.
+        // This is mutually exclusive with having a `DebugId`, and we only care about `abs_path` here.
+        // If we have a `DebugId`, we are guaranteed to use artifact bundles, and to have found the
+        // file in the check up above already.
+        if let Some(file) = self.try_fetch_file_from_artifacts(&key).await {
+            return self.files_by_key.entry(key).or_insert(Ok(file)).clone();
+        }
+
+        // Otherwise: Do a (cached) API lookup for the `abs_path` + `DebugId`
+        if self.remote_artifacts.is_empty() {
+            // TODO: the endpoint should really support filtering files,
+            // and ideally only giving us a single file that matches, but right now it just gives
+            // us the whole list of artifacts
+            self.remote_artifacts = self.list_artifacts().await;
```

Comment:
It does support filtering by name, so maybe we just give it our 4-variants of `abs_path` and this should suffice?

Reply:
You're right. Just checked and it's doing a partial matching, so should be enough.

```py
condition = Q(name__icontains=query[0])
    for name in query[1:]:
        condition |= Q(name__icontains=name)
```